Index: Src/Languages/Ruby/ironruby/Builtins/RubyRegex.cs
===================================================================
--- Src/Languages/Ruby/ironruby/Builtins/RubyRegex.cs	(revision 22765)
+++ Src/Languages/Ruby/ironruby/Builtins/RubyRegex.cs	(working copy)
@@ -112,7 +112,7 @@
 
         public Match/*!*/ Match(MutableString/*!*/ input, int start) {
             ContractUtils.RequiresNotNull(input, "input");
-            return Match(input, start, input.Length - start);
+            return Match(input, start, input.GetCharCount() - start);
         }
 
         public Match/*!*/ Match(MutableString/*!*/ input, int start, int count) {
